pkg/scheduler/framework/types.go

```diff
@@ -46,25 +46,50 @@ type ActionType int64
 
 // Constants for ActionTypes.
 const (
-	Add    ActionType = 1 << iota // 1
-	Delete                        // 10
-	// UpdateNodeXYZ is only applicable for Node events.
-	UpdateNodeAllocatable // 100
-	UpdateNodeLabel       // 1000
-	UpdateNodeTaint       // 10000
-	UpdateNodeCondition   // 100000
-	UpdateNodeAnnotation  // 1000000
+	Add ActionType = 1 << iota
+	Delete
 
-	All ActionType = 1<<iota - 1 // 1111111
+	// UpdateNodeXYZ is only applicable for Node events.
+	// If you use UpdateNodeXYZ,
+	// your plugin's QueueingHint is only executed for the specific sub-Update event.
+	// It's better to narrow down the scope of the event by using them instead of just using Update event
+	// for better performance in requeueing.
+	UpdateNodeAllocatable
+	UpdateNodeLabel
+	UpdateNodeTaint
+	UpdateNodeCondition
+	UpdateNodeAnnotation
+
+	// UpdatePodXYZ is only applicable for Pod events.
+	// If you use UpdatePodXYZ,
+	// your plugin's QueueingHint is only executed for the specific sub-Update event.
+	// It's better to narrow down the scope of the event by using them instead of Update event
+	// for better performance in requeueing.
+	UpdatePodLabel
+	// UpdatePodRequest is a update for pod's resource request calculated by resource.PodRequests() function.
+	UpdatePodRequest
+	// UpdatePodOther is a update for pod's other fields.
+	// NOT RECOMMENDED using it in your plugin's EventsToRegister,
+	// use Pod/Update instead when you have to subscribe Pod updates which are not covered by other UpdatePodXYZ events.
+	// Otherwise, your plugin might be broken when the upstream supports a new Pod specific Update event.
+	// It's used only for the internal event handling.
+	UpdatePodOther
```

Comment:
Maybe it's worth keeping it unexported... Then users can only use `Update`. But it still fullfills the purpose of making the matching logic more intuitive.

Reply:
Agree. Done 6fbe7714357b872c3b65b579e9eda7eb8f603d1c, but it requires another big diff to move pkg/scheduler/internal/queue/events.go under the framework package. 
(let me know if you prefer to do it in another follow-up PR, rather making this PR bigger.)

Reply:
(I had to force-push to solve a conflict made by the last commit.)